rviz_common/test/visualizer_app_test.cpp

```diff
@@ -57,3 +57,12 @@ TEST(VisualizerApp, shutdownsRosDuringExit) {
     rviz_common::VisualizerApp visualizer_app(std::move(ros_client));
   }
 }
+
+TEST(VisualizerApp, handlesEmptyFrameDuringExit) {
+  std::unique_ptr<rviz_common::ros_integration::RosClientAbstractionIface> ros_client =
+    std::make_unique<MockRosNodeStorage>();
+  EXPECT_CALL(*dynamic_cast<MockRosNodeStorage *>(ros_client.get()), shutdown()).Times(1);
+  rviz_common::VisualizerApp visualizer_app(std::move(ros_client));
+  // Simulate that frame_ is nullptr during exit, ensuring no crash occurs
+  visualizer_app.getRenderWindow();
```

Comment:
should you also check that this returns `nullptr`?

Reply:
https://github.com/ros2/rviz/pull/1359/commits/132f758995c6166f490ab36c6dbdb8e602f03e8e